src/main/java/core/basesyntax/service/impl/FruitShopDataProcessorService.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.handler.ShopOperationHandler;
+import core.basesyntax.service.DataProcessorService;
+import core.basesyntax.strategy.ShopOperationStrategy;
+import java.util.List;
+
+public class FruitShopDataProcessorService implements DataProcessorService {
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String COMMA = ",";
+    private final ShopOperationStrategy shopOperationStrategy;
+
+    public FruitShopDataProcessorService(ShopOperationStrategy shopOperationStrategy) {
+        this.shopOperationStrategy = shopOperationStrategy;
+
+    }
+
+    @Override
+    public void processData(List<String> data) {
+        for (String temp : data) {
```

Comment:
```suggestion
        for (String line : data) {
```
data is also plural form for datum, but it's rarely used

Reply:
Done